pkg/operators/kubenetworkmanager/kubenetworkmanager.go

```diff
@@ -0,0 +1,287 @@
+
+const (
+	RemoteKindPod     RemoteKind = "pod"
+	RemoteKindService RemoteKind = "svc"
+	RemoteKindOther   RemoteKind = "other"
+)
+
+// TODO: More granualar and "shareable" interfaces
+type AdditionalPodNetworkInformation interface {
+	// Local/Host
+	SetPodOwner(string)
+	SetPodHostIP(string)
+	SetPodIP(string)
+	SetPodLabels(map[string]string)
+
+	// Remote
+	GetRemoteIP() string
+	SetRemoteNamespace(string)
+	SetRemoteKind(string)
+	SetRemotePodLabels(map[string]string)
+}
+
+type KubeNetworkManager struct {
+	clientset *kubernetes.Clientset
+}
+
+func (k *KubeNetworkManager) Name() string {
+	return "KubeNetworkManager"
+}
+
+func (k *KubeNetworkManager) Description() string {
+	return "KubeNetworkManager handles more descriptive informations about remote ips"
+}
+
+func (k *KubeNetworkManager) GlobalParamDescs() params.ParamDescs {
+	return nil
+}
+
+func (k *KubeNetworkManager) ParamDescs() params.ParamDescs {
+	// TODO: Write patch so empty paramsDescs is okay
+	return params.ParamDescs{
+		{
+			Key:         "asdfsadf",
+			Description: "panics without at least one param",
+		},
+	}
+}
+
+func (k *KubeNetworkManager) Dependencies() []string {
+	return []string{"KubeManager"}
+}
+
+func (k *KubeNetworkManager) CanOperateOn(gadget gadgets.GadgetDesc) bool {
+	km := kubemanager.KubeManager{}
+	if !km.CanOperateOn(gadget) {
+		return false
+	}
+
+	gi, ok := gadget.(gadgets.GadgetInstantiate)
+	if !ok {
+		return false
+	}
+
+	instance, err := gi.NewInstance()
+	if err != nil {
+		log.Printf("failed to create dummy instance")
+		return false
+	}
+
+	_, hasResolver := instance.(SetIPToK8sResolverInterface)
+	_, hasAdditionalInf := gadget.EventPrototype().(AdditionalPodNetworkInformation)
+
+	return hasResolver && hasAdditionalInf
+}
+
+func (k *KubeNetworkManager) Init(params *params.Params) error {
+	clientset, err := k8sutil.NewClientset("")
+	if err != nil {
+		return fmt.Errorf("creating new k8s clientset: %w", err)
+	}
+	k.clientset = clientset
+	return nil
+}
+
+func (k *KubeNetworkManager) Close() error {
+	return nil
+}
+
+func (k *KubeNetworkManager) Instantiate(gadgetContext operators.GadgetContext, gadgetInstance any, params *params.Params) (operators.OperatorInstance, error) {
+	traceInstance := &KubeNetworkManagerInstance{
+		gadgetCtx: gadgetContext,
+		manager:   k,
+		gadget:    gadgetInstance,
+	}
+
+	return traceInstance, nil
+}
+
+type KubeNetworkManagerInstance struct {
+	gadgetCtx operators.GadgetContext
+	manager   *KubeNetworkManager
+	gadget    any
+}
+
+func (m *KubeNetworkManagerInstance) Name() string {
+	return "KubeNetworkManagerInstance"
+}
+
+func (m *KubeNetworkManagerInstance) PreGadgetRun() error {
+	resolver, hasResolver := m.gadget.(SetIPToK8sResolverInterface)
+	if hasResolver {
+		resolver.SetIPToK8sResolver(m.resolver)
+	}
+	return nil
+}
+
+func (m *KubeNetworkManagerInstance) PostGadgetRun() error {
+	return nil
+}
+
+func (m *KubeNetworkManagerInstance) resolver(ip string) string {
```

Comment:
Will update the PR later and split these operators into 2 for this PR.

The idea behind `SetIPToK8sResolverInterface` is to have a flexible way to implement #996 . This should be one separate operator.
The remaining enriching needed for `trace network` for know will be its own operator. Maybe we can see what parts of it can be useful for other gadgets as well

Reply:
If it works like before already, we should postpone adding the resolver function. I wouldn't create another operator for it though, as it's reusing the information that this operator is polling already. In general, the resolver function should be used if you have a couple of different IPs per event that you want to resolve (in maybe different ways).